gcp/src/main/java/org/apache/iceberg/gcp/biglake/BigLakeCatalog.java

```
@@ -0,0 +1,396 @@
+    if (purge) {
+      ops = newTableOps(identifier);
+      // TODO: to catch NotFoundException as in https://github.com/apache/iceberg/pull/5510.
+      lastMetadata = ops.current();
+    }
+
+    try {
+      client.deleteTable(tableName(dbId, identifier.name()));
+    } catch (NoSuchTableException e) {
+      return false;
+    }
+
+    if (ops != null && lastMetadata != null) {
+      CatalogUtil.dropTableData(ops.io(), lastMetadata);
+    }
+
+    return true;
+  }
+
+  @Override
+  public void renameTable(TableIdentifier from, TableIdentifier to) {
+    String fromDb = databaseId(from.namespace());
+    validateDatabaseId(fromDb, from.namespace());
+
+    String toDb = databaseId(to.namespace());
+    validateDatabaseId(toDb, to.namespace());
+
+    Preconditions.checkArgument(
+        fromDb.equals(toDb),
+        "Cannot rename table %s to %s: database must match",
+        from.toString(),
+        to.toString());
+    client.renameTable(tableName(fromDb, from.name()), tableName(toDb, to.name()));
+  }
+
+  @Override
+  public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+    if (namespace.isEmpty()) {
+      // Used by `CREATE NAMESPACE <catalog>`. Create a BLMS catalog linked with Iceberg catalog.
+      client.createCatalog(catalogName, Catalog.getDefaultInstance());
+      LOG.info("Created BigLake catalog: {}", catalogName.toString());
+    } else if (namespace.levels().length == 1) {
+      // Create a database.
+      String dbId = databaseId(namespace);
+      Database.Builder builder = Database.newBuilder().setType(Database.Type.HIVE);
+      builder
+          .getHiveOptionsBuilder()
+          .putAllParameters(metadata)
+          .setLocationUri(databaseLocation(dbId));
+
+      client.createDatabase(DatabaseName.of(projectId, region, catalogId, dbId), builder.build());
+    } else {
+      throw new IllegalArgumentException(
+          String.format("Invalid namespace (too long): %s", namespace));
```

Prefer throwing exceptions at the start of methods during validation, rather than mixing validation and core logic.

---

Done.